lib/ssl/src/ssl_cipher.erl

```diff
@@ -1193,39 +1181,34 @@ ec_keyed(_) ->
 
 %% Certs key is an ECC key
 ec_keyed_suites(Ciphers) ->
-    filter_suites(Ciphers, #{key_exchange_filters => [fun(Kex) -> ec_keyed(Kex) end],
-                             cipher_filters => [],
-                             mac_filters => [],
-                             prf_filters => []}).
+    ec_generic_suites(Ciphers, fun(Kex) -> ec_keyed(Kex) end).
 
 %% EC Certs key usage keyAgreement
 ec_ecdh_suites(Ciphers)->
-    filter_suites(Ciphers, #{key_exchange_filters => [fun(ecdh_ecdsa) -> true;
-                                                         (_) -> false
-                                                      end],
-                             cipher_filters => [],
-                             mac_filters => [],
-                             prf_filters => []}).
+    ec_generic_suites(Ciphers, fun(ecdh_ecdsa) -> true;
+                                  (_) -> false
+                               end).
 
 %% EC Certs key usage digitalSignature
 ec_ecdhe_suites(Ciphers) ->
-    filter_suites(Ciphers, #{key_exchange_filters => [fun(ecdhe_ecdsa) -> true;
-                                                         (ecdhe_rsa) -> true;
-                                                         (_) -> false
-                                                      end],
-                             cipher_filters => [],
-                             mac_filters => [],
-                             prf_filters => []}).
+    ec_generic_suites(Ciphers, fun(ecdhe_ecdsa) -> true;
+                                  (ecdhe_rsa) -> true;
+                                  (_) -> false
+                               end).
 %% RSA Certs key usage digitalSignature
 rsa_ecdhe_dhe_suites(Ciphers) ->
-    filter_suites(Ciphers, #{key_exchange_filters => [fun(dhe_rsa) -> true;
-                                                         (ecdhe_rsa) -> true;
-                                                         (_) -> false
-                                                      end],
+    ec_generic_suites(Ciphers, fun(dhe_rsa) -> true;
+                                  (ecdhe_rsa) -> true;
+                                  (_) -> false
+                               end).
+
+ec_generic_suites(Ciphers, Fn) ->
+    filter_suites(Ciphers, #{key_exchange_filters => [Fn],
```

Comment:
I think ec_generic_suites is a confusing name here as that only depends on the input funs. Maybe call the function filter_keyexcahnge or maybe filter_kex. Could then be used also in a few more places.